llvm/include/llvm/IR/DataLayout.h

```diff
@@ -361,6 +414,16 @@ class DataLayout {
     return PTy && isNonIntegralPointerType(PTy);
   }
 
+  bool shouldAvoidPtrToInt(Type *Ty) const {
+    auto *PTy = dyn_cast<PointerType>(Ty);
+    return PTy && shouldAvoidPtrToInt(PTy->getPointerAddressSpace());
```

Comment:
It seems odd to ask about ptrtoint for something where you don't know it's a pointer already, but I guess this is to match isNonIntegralPointerType which seems to have a decent number of uses.

Reply:
Yeah I kept this check to allow replacing existing helper calls.

Reply:
This also needs to handle the vector of pointer case